src/sentry/api/helpers/deprecation.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import functools
+from datetime import datetime, timedelta
+
+from croniter import croniter
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.status import HTTP_410_GONE as GONE
+
+from sentry.utils.settings import is_self_hosted
+
+BROWNOUT_LENGTH = timedelta(days=30)
+GONE_MESSAGE = {"message": "This API no longer exists."}
+DEPRECATION_HEADER = "X-Sentry-Deprecation-Date"
+SUGGESTED_API_HEADER = "X-Sentry-Replacement-Endpoint"
+
+# TODO: Make these configurable from redis
```

Comment:
You could use `sentry.options` for this. Ops has tooling to update options already.

Reply:
Yeah, I was going to do more of the brownout logic in a followup PR. I need to think about what sort of format the brownout schedule should be.